libsolidity/interface/StandardCompiler.cpp

```diff
@@ -292,10 +292,27 @@ boost::optional<Json::Value> checkSettingsKeys(Json::Value const& _input)
 
 boost::optional<Json::Value> checkOptimizerKeys(Json::Value const& _input)
 {
-	static set<string> keys{"enabled", "runs"};
+	static set<string> keys{"details", "enabled", "runs"};
 	return checkKeys(_input, keys, "settings.optimizer");
 }
 
+boost::optional<Json::Value> checkOptimizerDetailsKeys(Json::Value const& _input)
+{
+	static set<string> keys{"peephole", "jumpdestRemover", "orderLiterals", "deduplicate", "cse", "constantOptimizer", "yul", "yulDetails"};
+	return checkKeys(_input, keys, "settings.optimizer.details");
+}
+
+boost::optional<Json::Value> checkOptimizerDetail(Json::Value const& _details, std::string const& _name, bool& _setting)
+{
+	if (_details.isMember(_name))
+	{
+		if (!_details[_name].isBool())
+			return formatFatalError("JSONError", "\"settings.optimizer.details." + _name + "\" must be Boolean");
```

Comment:
Lowercase `b` in boolean.

Reply:
I would actually disagree: https://en.wikipedia.org/wiki/Boolean_data_type

Reply:
The rest of the file refers to it with lowercase

Reply:
Not true: https://github.com/ethereum/solidity/blob/0e02de73f74af864a9ea1a4f39e919287d966ab7/libsolidity/interface/StandardCompiler.cpp#L319 :)